### ISSUE TITLE
Changing currency code in Localization Configuration

### ISSUE THREAD

**Issue by wd-code** [opensourcepos/opensourcepos#2574]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.3.0
- OSPOS git commit hash is: 0b9a76
- PHP version is: 7.3
- MySQL or MariaDB version is: MarisDB 5.5
- OS and version is: Debian 9
- WebServer is: Nginx 1.14.1
- Selected language is: French / English

### Issue / Bug
![image](https://user-images.githubusercontent.com/6681481/66261455-dcc94280-e7cd-11e9-9849-cf71378d5921.png)

In the Localizationconfiguration, when I want to change the currency code from USD to EUR then hit the submit button, the currency code comes back to USD.

**Comment by SteveIreland** [comment]
I'll take a look a this issue.

**Comment by SteveIreland** [comment]
I don't believe that the currency code does anything of value at this time.  What is it that you are expecting the currency code to do for you?

Regardless, there is some odd behavior that can and should be cleaned up.  I'll add this to the  short list of tasks I have for 3.3.1.  I encountered this issue when I was working on a change for alternate currency codes that was and never will be merged.  The odd behavior is not a show stopper for any.  For now, after changing the localization code be sure to tab out to have the changes applied and then you should be able to apply the overrides.

**Comment by wd-code** [comment]
> I don't believe that the currency code does anything of value at this time. What is it that you are expecting the currency code to do for you?

I don't think so, but since It seem to not work I just put this here.

> For now, after changing the localization code be sure to tab out to have the changes applied and then you should be able to apply the overrides.

Just try the trick, so it work when I tab out, but not when I "focus" out it dosen't. Is it a bit weird ?

**Comment by izeblood** [comment]
It only works mine in 3 attempts....

**Comment by SteveIreland** [comment]
I agree it is weird, but that's the bug that I will be fixing. It seems to be that the ajax method that establishes the defaults isn't executed if you instead go direct to save - but I haven't yet confirmed that.

**Comment by daN4cat** [comment]
@SteveIreland when you have fixed this issue, I would make it the last for 3.3.1 unless there is some other big issue. We will give some time before releasing.
Also to make sure translations are fine.

**Comment by SteveIreland** [comment]
Sounds good to me.

**Comment by SteveIreland** [comment]
**Progress report**:  I completed the coding and the initial tests look pretty good.

So my plans are to do a bit more testing this evening and to submit a pull request for it sometime tonight.

_To think about:_ The localization code establishes the default values for currency symbol and currency code but only if the currency symbol and currency code are left blank - otherwise the values for currency symbol will override the default.  The currency code no longer plays a role in the configuration other than being loaded with a value that is established by the localization code.

I hope that makes sense.

**Comment by daN4cat** [comment]
@wd-code can you please test latest code from master and confirm it's fine? Thanks

**Comment by SteveIreland** [comment]
Thank you @hbabuu for taking a look at it.  Odd situation.  I downloaded the latest master and ran a test.  I was able to replicate the error you found in the first 2 or 3 tests I ran and then all of the sudden it started working.  Similar to what @izeblood experienced.

There is something else going on that I haven't yet stumbled across, but another fix is in the making.

**Comment by SteveIreland** [comment]
I have another pull request in place to try to fix this issue.

It turns out that there was a timing issue in the use of jQuery to populate the values for the third party utility we use for form validation.  Fortunately there was an example in another form that showed how to do it properly so I was able to learn from that.

I also took this opportunity to change it around a bit so that the changes were a little more clear and hopefully are more logical.  If you change the locale then the default values for currency symbol and currency code for that locale are immediately applied and shown to the user.  Then the user can override the defaults.  I feel that is much smoother that what was previously in place.

Let me know if you disagree.
